lib/utils.pm

```diff
@@ -510,6 +510,18 @@ sub sle_version_at_least {
     die "unsupported SLE $version_variable $version in check";
 }
 
+# To cope with staging version naming and this method should only be used
+# to leap_version_at_least. This method returns 1 if it is a valid staging
+# naming or the version is matched to the one in settings.
+sub leap_staging_version_in_settings {
```

Comment:
I don't understand the name of the method or why you saw the need to implement a Leap specific version. Why does the same approach not apply for TW or SLE?

Reply:
we don't have *_version_at_least involved for TW; for SLE staging, `VERSION` in settings all are `15`, however in Leap Staging, `VERSION` is `15.0:S:A`, `15.0:S:A:B` and so on in openqa, therefor leap_version_at_least('15.0') at https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/installation/partitioning_lvm.pm#L103 doesn't work at all for Leap staging.

So the solution can be 1) tweak rsync.pl+openqa settings for all leap's medium type+openqa jobgroup, or 2) extract `VERSION` in settings to get the right version number of product. https://progress.opensuse.org/issues/27451

This PR is about the 2).

Reply:
@coolo?